opentelemetry-api/src/opentelemetry/_events/__init__.py

```diff
@@ -0,0 +1,196 @@
+from abc import ABC, abstractmethod
+from logging import getLogger
+from os import environ
+from typing import Any, Optional, cast
+
+from opentelemetry._logs import LogRecord
+from opentelemetry._logs.severity import SeverityNumber
+from opentelemetry.environment_variables import (
+    _OTEL_PYTHON_EVENT_LOGGER_PROVIDER,
+)
+from opentelemetry.trace.span import TraceFlags
+from opentelemetry.util._once import Once
+from opentelemetry.util._providers import _load_provider
+from opentelemetry.util.types import Attributes
+
+_logger = getLogger(__name__)
+
+
+class Event(LogRecord):
+
+    def __init__(
+        self,
+        name: str,
+        timestamp: Optional[int] = None,
+        trace_id: Optional[int] = None,
+        span_id: Optional[int] = None,
+        trace_flags: Optional["TraceFlags"] = None,
+        payload: Optional[Any] = None,
+        severity_number: Optional[SeverityNumber] = None,
+        attributes: Optional[Attributes] = None,
+    ):
+        super().__init__(
+            timestamp=timestamp,
+            trace_id=trace_id,
+            span_id=span_id,
+            trace_flags=trace_flags,
+            body=payload,  # type: ignore
+            severity_number=severity_number,
+            attributes=attributes,
+        )
+        self.name = name
+
+
+class EventLogger(ABC):
```

Comment:
Is there a reason that some classes are inherited from their logs counterpart and some others inherits from ABC?

Reply:
In the semantic convention (https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/events.md) it said: "In OpenTelemetry, Events are implemented as a specific type of LogRecord that conforms to the conventions included here, and Events [have their own API](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/event-api.md). The API abstracts away knowledge of LogRecord so that users only deal with Event semantics" so I chose to inherit LogRecord for the Event class. I could do the same for the EventLogger class as well if you would like me to do so.